packages/runtime/test/integration/deploy-asa.ts

```
@@ -35,4 +37,27 @@ describe("Deploy ASA with mutiple opt-in accounts", function () {
 		assert.isDefined(bob.getAssetHolding(asset));
 		assert.isDefined(alice.getAssetHolding(asset));
 	});
+
+	it("Should throw error when ASA definition not found in asa.yaml", () => {
+		expectRuntimeError(
+			() =>
+				runtime.deployASA("asa-invalid", {
+					creator: { ...john.account, name: "john" },
+				}),
+			RUNTIME_ERRORS.ASA.ASA_DEFINITION_NO_FOUND_IN_ASA_FILE
+		);
+	});
+
+	describe("ASA file is undefinded", function () {
+		useFixture("loop"); // project don't have asa-file
+		it("Should fail b/c we tried to get asa from asa file for deploy", () => {
```

I think we should you a convention as above. `Should throw an error when...`